pkg/grpc/marshaler.go

```diff
@@ -0,0 +1,81 @@
+package grpc
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// Deprecated: customMarshaler is a hack for keeping backward compatibility for duration objects.
+// TODO(ROX-25678): This should be deleted in 4.8.
+type customMarshaler struct {
+	*runtime.JSONPb
+}
+
+func (c customMarshaler) Unmarshal(data []byte, v interface{}) error {
+	unmarshalError := protojson.Unmarshal(data, v.(proto.Message))
+	if unmarshalError == nil {
+		return nil
+	}
+	return c.unmarshalBackwardCompatible(data, v, unmarshalError)
```

Comment:
nit:
That's weird to pass `unmarshalError` only to get it back.

`c.unmarshalBackwardCompatible` could return `bool` on error.